src/zip_code_tree.hpp

```diff
@@ -405,114 +426,231 @@ class ZipCodeForest {
-    /// which refers to the orientation in the snarl tree
-    /// This should run in linear time, but it is dependent on the values being sorted on to have a small range
-    /// min_ and max_value are the minimum and maximum value being sorted on
-    void radix_sort_zipcodes(vector<size_t>& zipcode_sort_order, const vector<sort_value_t>& sort_values_by_seed,
-                             const interval_and_orientation_t& interval, bool reverse_order,
-                             size_t min_value, size_t max_value) const; 
+    /// This stores information about the state of the forest as we fill it in
+    struct forest_growing_state_t {
 
-    /// Helper function to sort the seeds using std::sort
-    /// Sorts the slice of seeds in the given interval of zipcode_sort_order, which is a vector of indices
-    /// into seeds
-    void default_sort_zipcodes(vector<size_t>& zipcode_sort_order, const vector<sort_value_t>& sort_values_by_seed,
-                               const interval_and_orientation_t& interval, bool reverse_order) const; 
+        const vector<Seed>* seeds;
+
+        const SnarlDistanceIndex* distance_index;
+
+        vector<size_t> seed_sort_order;
+
+
+        //This stores the sort value and code type of each seed 
+        //This will change as forest building progresses but it will be set for the relevant seed 
+        //immediately before sorting
+        //The values also get used to calculate distance, as long as they have been set for the 
+        //correct depth
+        vector<sort_value_t> sort_values_by_seed;
+
+        //Stores the previous things of the current structure at each depth
+        //The children are stored at the depth of their parents. For example, for a root chain,
+        //the vector at index 0 would have the chain start, seeds that are on the chain, and the 
+        //start of snarls on the chain. Similarly, for a top-level snarl, at depth 1, the second 
+        //vector would contain the starts of chains at depth 2 
+        vector<vector<child_info_t>> sibling_indices_at_depth;
+
+        // We build a forest of trees. A new tree is formed either when a new top-level chain is 
+        // found (or a slice of a top-level chain if it is far enough away from the previous thing 
+        // in the chain), or when part of a chain in a snarl is too far from everything else in the
+        // snarl. In the second case, the entire subtree is found before determining that it should 
+        // be a subtree, and then it is copied into a new zip_tree_t in the forest.
+        // So only one tree is actively being added to at a time.
+        // This keeps track of which is the active tree, as an index into trees
+        // Note that this can't be an actual pointer to the forest because the address may move if 
+        // the vectors get shifted around in memory.
+        size_t active_zip_tree_i;
```

Comment:
Is the `_i` doing any useful work here?

Reply:
It's just so I know that it's an index into the vector of trees, not the tree itself. What I really wanted was a `ZipCodeTree& active_zip_tree`, but since the trees are stored in a vector, the pointers can move around

Reply:
I might say `_index`? Maybe just `active_index`? I guess `_i` means the same thing and is shorter but I don't know if we name anything else like that in the codebase.